8292691: Move CompilerConfig::is_xxx() inline functions out of compilerDefinitions.hpp

iklam opened the pull request:
Many of the `CompilerConfig::is_xxx()` inline functions in compilerDefinitions.hpp depend on the following headers

- compiler_globals.hpp
- c1_globals.hpp
- c2_globals.hpp
- jvmci_globals.hpp

However, only a few files actually use these functions. To improve C++ compilation time, we should move these inline functions to compilerDefinitions.inline.hpp

This reduces the inclusion of the above headers from 762 to 321.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292691](https://bugs.openjdk.org/browse/JDK-8292691): Move CompilerConfig::is_xxx() inline functions out of compilerDefinitions.hpp


### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [adda27ae](https://git.openjdk.org/jdk/pull/9953/files/adda27ae445eb61eaf81f18b61c403d29ceb167b)
 * [Igor Veresov](https://openjdk.org/census#iveresov) (@veresov - **Reviewer**)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9953/head:pull/9953` \
`$ git checkout pull/9953`

Update a local copy of the PR: \
`$ git checkout pull/9953` \
`$ git pull https://git.openjdk.org/jdk pull/9953/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9953`

View PR using the GUI difftool: \
`$ git pr show -t 9953`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9953.diff">https://git.openjdk.org/jdk/pull/9953.diff</a>

</details>

vnkozlov reviewed (approved):
Looks fine.

veresov commented:
Many of these are called in pretty hot paths in `CompilationPolicy`. That is when an interpreter or a c1-compiled method is invoking the policy on method invocations and back branches. So, I would like to keep these inline.

Could we move these to `compilerDefinitions.inline.hpp` and include it only where it's necessary?

vnkozlov commented:
> Many of these are called in pretty hot paths in `CompilationPolicy`. That is when an interpreter or a c1-compiled method is invoking the policy on method invocations and back branches. So, I would like to keep these inline.
> 
> Could we move these to `compilerDefinitions.inline.hpp` and include it only where it's necessary?

Is `CompilationPolicy` code hot? It triggers only on counters overflow. It even less hot when methods are compiled by C2.
But I am fine if moving them to `compilerDefinitions.inline.hpp` reduce VM build time too.

veresov commented:
It is reasonably hot. The interpreter calls is it for every 128th invocation of a method and every 1024th back branch in a particular method. C1 calls it every 1024th invocation and 8192th back branch.

iklam commented:
> It is reasonably hot. The interpreter calls it for every 128th invocation of a method and every 1024th back branch in a particular method. C1 calls it every 1024th invocation and 8192th back branch.

I moved the inline functions to compilerDefinitions.inline.hpp. I had to add the inclusion of this file to a few source files, but most of them were missing an include of compilerDefinitions.hpp anyway.

I also moved CompilerConfig::is_interpreter_only() from compilerDefinitions.cpp into compilerDefinitions.inline.hpp, since it's called by some of the CompilerConfig::is_xxx() inline functions.

veresov reviewed (approved):
Thank you! Looks good.

iklam commented:
Thanks for the review @veresov and @vnkozlov 
I updated the issue title and description to reflect the final version.
/integrate
